cpp/src/io/utilities/block_utils.cuh

```diff
@@ -20,17 +20,52 @@
 namespace cudf {
 namespace io {
 #if (__CUDACC_VER_MAJOR__ >= 9)
```

Comment:
We don't support versions below 10.1, so this condition and the else branch can be removed.